java-symbolic-execution/src/main/java/org/sonar/java/se/checks/NonNullSetToNullCheck.java

```diff
@@ -143,15 +146,20 @@ private static boolean callsThisConstructor(MethodTree constructor) {
   }
 
   private void checkVariable(CheckerContext context, MethodTree tree, final Symbol symbol) {
-    String nonNullAnnotation = nonNullAnnotation(symbol);
-    if (nonNullAnnotation == null || nonNullAnnotation.startsWith("javax.validation.constraints.") || symbol.isStatic()) {
+    Optional<String> nonnullAnnotationAsString = getNonnullAnnotationAsString(symbol);
+    if (nonnullAnnotationAsString.isEmpty() || isJavaxValidationConstraint(symbol) || symbol.isStatic()) {
       return;
     }
     if (isUndefinedOrNull(context, symbol)) {
-      context.reportIssue(tree.simpleName(), this, MessageFormat.format("\"{0}\" is marked \"{1}\" but is not initialized in this constructor.", symbol.name(), nonNullAnnotation));
+      context.reportIssue(tree.simpleName(), this,
+        MessageFormat.format("\"{0}\" is marked \"{1}\" but is not initialized in this constructor.", symbol.name(), nonnullAnnotationAsString.get()));
     }
   }
 
+  private static boolean isJavaxValidationConstraint(Symbol symbol) {
+    return symbol.metadata().nullabilityData().annotation().symbol().type().fullyQualifiedName().startsWith("javax.validation.constraints.");
```

Comment:
Should we check that `symbol.metadata().nullabilityData().annotation()` is not null here?